--- /home/jacek/git/sha/brg_endian.h	2023-07-25 23:20:58.157487162 +0200
+++ brg_endian.h	2023-07-25 23:27:20.889797352 +0200
@@ -16,6 +16,7 @@
 and fitness for purpose.
 ---------------------------------------------------------------------------
 Issue Date: 20/12/2007
+Changes for ARM 9/9/2010 [Downstream relative to Gladman's GitHub, upstream to Qt]
 */
 
 #ifndef _BRG_ENDIAN_H
@@ -39,9 +40,9 @@
       defined( __CYGWIN32__ ) || defined( __DJGPP__ ) || defined( __osf__ )
 #  include <machine/endian.h>
 #elif defined( __linux__ ) || defined( __GNUC__ ) || defined( __GNU_LIBRARY__ )
-#  if !defined( __MINGW32__ ) && !defined( _AIX )
+#  if !defined( __MINGW32__ ) && !defined( _AIX ) && !defined(Q_OS_QNX)
 #    include <endian.h>
-#    if !defined( __BEOS__ )
+#    if !defined( __BEOS__ ) && !defined(Q_OS_RTEMS) && !defined(Q_OS_VXWORKS)
 #      include <byteswap.h>
 #    endif
 #  endif
@@ -119,12 +120,18 @@
       defined( THINK_C ) || defined( __VMCMS__ ) || defined( _AIX )
 #  define PLATFORM_BYTE_ORDER IS_BIG_ENDIAN
 
-#elif 0     /* **** EDIT HERE IF NECESSARY **** */
+#elif defined(__arm__)
+# ifdef __BIG_ENDIAN
+#  define PLATFORM_BYTE_ORDER IS_BIG_ENDIAN
+# else
+#  define PLATFORM_BYTE_ORDER IS_LITTLE_ENDIAN
+# endif
+#elif 1     /* **** EDIT HERE IF NECESSARY **** */
 #  define PLATFORM_BYTE_ORDER IS_LITTLE_ENDIAN
 #elif 0     /* **** EDIT HERE IF NECESSARY **** */
 #  define PLATFORM_BYTE_ORDER IS_BIG_ENDIAN
 #else
-#  error Please edit lines 126 or 128 in brg_endian.h to set the platform byte order
+#  error Please edit lines 129 or 131 in brg_endian.h to set the platform byte order
 #endif
 
 #endif
